aspnetcore-webapi/MediatrExceptionHandler/MediatrExceptionHandler/GlobalExceptionMiddleware.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace MediatrExceptionHandler
+{
+    using System.Diagnostics;
+    using System.Net;
+    using System.Text.Json;
+
+    public class GlobalExceptionMiddleware : IMiddleware
+    {
+        private readonly ILogger<GlobalExceptionMiddleware> _logger;
+
+        public GlobalExceptionMiddleware(ILogger<GlobalExceptionMiddleware> logger)
+        {
+            _logger = logger;
+        }
+
+        public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+        {
+            try
+            {
+                await next(context);
+            }
+            catch (Exception ex)
+            {
+                var exception = ex.Demystify();
+
+                _logger.LogError(exception, "An error ocurred: {Message}", exception.Message);
+
+                context.Response.ContentType = "application/json";
+                context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+
+                var result = JsonSerializer.Serialize(new { error = "A server error ocurred" });
```

Comment:
We can use `ProblemDetails `class instead from Microsoft. And also add the exception.Message as `Detail`.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.problemdetails?view=aspnetcore-7.0